Script/AtomicEditor/hostExtensions/languageExtensions/TypescriptLanguageExtension.ts

```diff
@@ -362,57 +372,122 @@ export default class TypescriptLanguageExtension extends Atomic.ScriptObject imp
-
-      // Build out the vscode tasks.json
-      const tasks = {
-         "command": "tsc",
-         "isShellCommand": true,
-         "args": ["-p", "."],
-         "showOutput": "always",
-         "problemMatcher": "$tsc"
-     };
-
-     const tasksDir = Atomic.addTrailingSlash(projectDir + ".vscode");
-     Atomic.fileSystem.createDir(tasksDir);
-     const tasksFile = new Atomic.File(tasksDir + "tasks.json", Atomic.FileMode.FILE_WRITE);
-     tasksFile.writeString(JSON.stringify(tasks, null, 4));
-     tasksFile.close();
+    generateExternalProject() {
+        const projectDir = ToolCore.toolSystem.project.projectPath;
+
+        // Create the typings folder in project root
+        const projectDirTypings = Atomic.addTrailingSlash(projectDir + "typings/ambient/atomicgameengine");
+        Atomic.getFileSystem().createDir(projectDirTypings);
+
+        // Copy the Atomic.d.ts definition file to the typings folder
+        const toolDataDir = ToolCore.toolEnvironment.toolDataDir;
+        const typescriptSupportDir = Atomic.addTrailingSlash(toolDataDir + "TypeScriptSupport");
+        Atomic.getFileSystem().copy(typescriptSupportDir + "Atomic.d.ts", projectDirTypings + "Atomic.d.ts");
+
+        // Generate a tsconfig.json file
+        if (this.isTypescriptProject) {
+            const tsconfigFile = new Atomic.File(projectDir + "tsconfig.json", Atomic.FileMode.FILE_WRITE);
+            let tsconfig = {
+                compilerOptions: defaultCompilerOptions
+            };
+
+            // Don't use fully qualified path in the persistent tsconfig file, just use a relative path from the tsconfig
+            tsconfig.compilerOptions["baseUrl"] = "./Resources";
+            tsconfig.compilerOptions.allowJs = false;
 
+            tsconfigFile.writeString(JSON.stringify(tsconfig, null, 4));
+            tsconfigFile.close();
+        }
+
+        this.generateVsCodeFiles();
     }
+
+    /**
+     * Generates the extra vscode files for launching the debugger, the player, etc
+     */
+    generateVsCodeFiles() {
+        const projectDir = ToolCore.toolSystem.project.projectPath;
+        const vscodeDir = Atomic.addTrailingSlash(projectDir + ".vscode");
+        Atomic.fileSystem.createDir(vscodeDir);
+
+        // Build out the vscode tasks.json
+        const tasks = {
+            "version": "0.1.0",
+            "tasks": [
+                {
+                    "taskName": "Build",
+                    "command": "tsc",
+                    "isShellCommand": true,
+                    "args": [
+                        "-p",
+                        "."
+                    ],
+                    "showOutput": "always",
+                    "problemMatcher": "$tsc",
+                    "isBuildCommand": true
+                },
+                {
+                    "taskName": "Debug Atomic Player",
+                    "command": `${ToolCore.toolEnvironment.editorBinary}`,
+                    "args": [
```

Comment:
Cool, love the interaction between VSCode and Atomic's ToolCore 👍, looking forward to having a look at VSCode + NET Core + Atomic...